packages/expect/src/types.ts

```diff
@@ -85,17 +85,25 @@ export type AsyncExpectationResult = Promise<SyncExpectationResult>
 
 export type ExpectationResult = SyncExpectationResult | AsyncExpectationResult
 
-export interface RawMatcherFn<T extends MatcherState = MatcherState> {
-  (this: T, received: any, ...expected: Array<any>): ExpectationResult
+export interface RawMatcherFn<T extends MatcherState = MatcherState, E extends Array<any> = Array<any>> {
+  (this: T, received: any, ...expected: E): ExpectationResult
 }
 
+// Allow unused `T` to preserve its name for extensions.
+// Type parameter names must be identical when extending those types.
+// eslint-disable-next-line
+export interface Matchers<T = any> {}
+
 export type MatchersObject<T extends MatcherState = MatcherState> = Record<
   string,
   RawMatcherFn<T>
-> & ThisType<T>
+> & ThisType<T> & {
+  [K in keyof Matchers<T>]?: RawMatcherFn<T, Parameters<Matchers<T>[K]>>
+}
 
 export interface ExpectStatic
   extends Chai.ExpectStatic,
+  Matchers,
```

Comment:
I've learned extending `AsymmetricMatchersContaining` on the `vitest` package level wasn't enough for TS to pick up the custom matchers when used on the `expect` object statically: `expect.toBeMyThing()`. That's because the `ExpectStatic` for the object comes from `@vitest/expect`, not `vitest`. I believe this should be the right way to extend it. 

I also tried making `AsymmetricMatchersContaining` to extend custom `Matchers` instead but that did not result in the custom matcher types to be present on `ExpectStatic`.